### ISSUE TITLE
The function name 'unique' is a little misleading

### ISSUE THREAD

**Issue by hiiwave** [Dobiasd/FunctionalPlus#66]
I'm glad to just find this great library! After some exploration, I found that the function name `unique` is a little misleading. Here `unique([1,2,2,3,2]) == [1,2,3,2]`, though I would expect `unique([1,2,2,3,2]) == [1,2,3]` as the behavior of [`numpy.unique` ](https://docs.scipy.org/doc/numpy/reference/generated/numpy.unique.html)in python. 

Despite this name is followed from `std::unique`, which remove consecutive duplicates in range, I personally think using a slight different name (e.g. `unique_consecutime`) here would be beneficial for avoiding misuse. However it's just personal thought. Please feel free to take it or leave it :)

**Comment by Dobiasd** [comment]
First, thank you for looking trough the library and caring about its improvement.

I had a similar thought process when implementing `unique` and `nub`. I decided to name this function `unique` because I made similar compromises with other function names, e.g. `transform` vs. `map`. I target developers being more familiar with C++ than with functional programming in general and thus sometimes try to keep the stl names.

Also I hesitate to make an API-incompatible change because of this problem. :wink:

However to avoid future confusion, I borrowed a sentence from [cppreference](http://en.cppreference.com/w/cpp/algorithm/unique) to [add a bit more documentation to `fplus::unique`](https://github.com/Dobiasd/FunctionalPlus/commit/a570f168b74df6512344ac66ede17cd15775633b). I hope this makes things more clear.
